07_deploy_masters.sh

```diff
@@ -20,45 +20,33 @@ if [ $(sudo podman ps | grep -w -e "ironic$" -e "ironic-inspector$" | wc -l) !=
     exit 1
 fi
 
-# Clean previously env
-nodes=$(openstack baremetal node list)
-for node in $(jq -r .nodes[].name ${MASTER_NODES_FILE}); do
-  if [[ $nodes =~ $node ]]; then
-    openstack baremetal node undeploy $node --wait || true
-    openstack baremetal node delete $node
-  fi
-done
+mkdir ocp/tf-master
+cp ocp/master.ign ocp/tf-master
 
-openstack baremetal create $MASTER_NODES_FILE
-mkdir -p configdrive/openstack/latest
-cp ocp/master.ign configdrive/openstack/latest/user_data
-for node in $(jq -r .nodes[].name $MASTER_NODES_FILE); do
+cat > ocp/tf-master/main.tf <<EOF
+provider "ironic" {
+  "url" = "http://localhost:6385/v1"
+  "microversion" = "1.50"
+}
+EOF
 
-  # FIXME(shardy) we should parameterize the image
-  openstack baremetal node set $node --instance-info "image_source=http://172.22.0.1/images/$RHCOS_IMAGE_FILENAME_LATEST" --instance-info image_checksum=$(curl http://172.22.0.1/images/$RHCOS_IMAGE_FILENAME_LATEST.md5sum) --instance-info root_gb=25 --property root_device="{\"name\": \"$ROOT_DISK\"}"
-  openstack baremetal node manage $node --wait
-  openstack baremetal node provide $node --wait
-done
+IMAGE_SOURCE="http://172.22.0.1/images/$RHCOS_IMAGE_FILENAME_LATEST"
+IMAGE_CHECKSUM=$(curl http://172.22.0.1/images/$RHCOS_IMAGE_FILENAME_LATEST.md5sum)
 
-for node in $(jq -r .nodes[].name $MASTER_NODES_FILE); do
-  openstack baremetal node deploy --config-drive configdrive $node
-done
+ROOT_GB="25"
+ROOT_DEVICE="/dev/vda"
 
-# Note we have to tolerate failure with the || echo 0 due to this issue
-# https://storyboard.openstack.org/#!/story/2005093
-NUM_ACTIVE=$(openstack baremetal node list --fields name --fields provision_state | grep master | grep active | wc -l || echo 0)
-while [ "$NUM_ACTIVE" != "3" ]; do
-  if openstack baremetal node list --fields name --fields provision_state | grep master | grep -e error -e failed; then
-    openstack baremetal node list
-    echo "Error detected waiting for baremetal nodes to become active" >&2
-    exit 1
-  fi
-  sleep 10
-  NUM_ACTIVE=$(openstack baremetal node list --fields name --fields provision_state | grep master | grep active | wc -l || echo 0)
+for i in $(seq 0 2); do
+  master_node_to_tf $i $IMAGE_SOURCE $IMAGE_CHECKSUM $ROOT_GB $ROOT_DEVICE >> ocp/tf-master/main.tf
 done
 
+echo "Deploying master nodes"
+cd ocp/tf-master
+terraform init
+terraform apply --auto-approve
```

Comment:
Could we enable debug logging as well? 

`export TF_LOG=DEBUG`

Reply:
Good call!